app/[addressOrDomain]/page.tsx

```diff
@@ -189,18 +197,144 @@ export default function Page({ params }: AddressOrDomainProps) {
+    for await (const userDapp of userDapps) {
+      const positions = userDapp.products[0]?.positions;
+      if (!positions) continue;
+
+      for (const position of positions) {
+        const totalBalances = position.totalBalances;
+        for (const tokenAddress of Object.keys(totalBalances)) {
+          const tokenBalance = totalBalances[tokenAddress];
+          const tokenInfo = tokens[tokenAddress];
+          if (!tokenInfo) continue;
+          if (tokenBalance === "0") continue;
+
+          const value = await calculateTokenPrice(
+            tokenAddress,
+            tokenToDecimal(tokenBalance, tokenInfo.decimals),
+            "USD"
+          );
+
+          totalProtocolValue += value;
+
+          // Don't add negative balances to total value (or may have percentages > 100%)
+          if (value < 0) continue;
+          // Add to total value regardless of protocol
+          totalValue += value;
+          console.log('Added protocol value:', value, 'from token:', tokenInfo.symbol, 'protocol:', dapps[userDapp.dappId].name);
+          // Only add to asset breakdown if not from excluded protocol. Restaking anyone?
+          if (tokenInfo.dappId && EXCLUDED_PROTOCOL_IDS.includes(tokenInfo.dappId)) {
+            continue;
+          }
+          const symbol = tokenInfo.symbol || "Unknown";
+          assetValues[symbol] = (assetValues[symbol] || 0) + value;
+        }
+      }
+    }
+
+    console.log("!!!Total value:", totalValue, "Total token value:", totalTokenValue, "Total protocol value:", totalProtocolValue);
+
+    // Convert to percentages and format
+    const sortedAssets = Object.entries(assetValues)
+      .sort(([, a], [, b]) => b - a)
+      .map(([symbol, value]) => ({
+        itemLabel: symbol,
+        itemValue: ((value / totalValue) * 100).toFixed(2),
+        itemValueSymbol: "%",
+        color: "" // Colors will be assigned later
+      }));
+
+    console.log(JSON.stringify(sortedAssets, null, 2));
+    console.log(sortedAssets.slice());
+
+    // Handle "Others" category if needed
+    if (sortedAssets.length > 4) {
+      const others = sortedAssets.slice(4).reduce(
+        (sum, asset) => sum + parseFloat(asset.itemValue),
+        0
+      );
+      sortedAssets.splice(4);
+      sortedAssets.push({
+        itemLabel: "Others",
+        itemValue: others.toFixed(2),
+        itemValueSymbol: "%",
+        color: ""
+      });
+    }
+
+    // Assign colors
+    const colors = ["#1E2097", "#637DEB", "#2775CA", "#5CE3FE", "#F4FAFF"];
+    sortedAssets.forEach((asset, index) => {
+      asset.color = colors[index];
+    });
+
+    return sortedAssets;
+  };
+
+  const fetchPortfolioAssets = useCallback(async (data: {
+    dapps: ArgentDappMap,
+    tokens: ArgentTokenMap,
+    userTokens: ArgentUserToken[],
+    userDapps: ArgentUserDapp[]
+  }) => {
+    const { dapps, tokens, userTokens, userDapps } = data;
+    try {
+      // TODO: correct this condition
```

Comment:
What do you mean ?

Reply:
Nothing, just a misplaced comment, was intended for my code, already deleted

Reply:
Also, console logs are deleted too

Reply:
Removed useless comments and logs